### DISCUSSION TITLE
New Function: Battery Device

### DISCUSSION THREAD

**Discussion by Parateam** [freakent/dbus-mqtt-devices#127] [Ideas]
Hello Martin, i have 2 Lynx distributors in my Setup. They have an i2C port for the fuse status.

Now i want to report blown fuses in the GX and VRM. Normally this is only possible with a shount or battary monitor.

My idea is to use an ESP32 to readout the i2C and publish to MQTT. This might be possible with an battery device.

Could you add this device type in your driver? I use your driver for all konds of Inverters via NodeRed.

Here is additional Information

https://github.com/twam/dbus-lynx-distributor

Thanks Peter

**Comment by freakent** [comment]
That's such a coincidence, I saw your post on Victron forum and was reading your GitHub project this morning. Nice job decoding the i2c interface, that can't have been easy. 

I noticed you were using battery as the device for this. How does the status of the fuses show up in your GX device? Is battery really the best device to represent this?

**Comment by Parateam** [comment]
Hello Martin, i think so. My research points to this. I think this is the
only way to get blown fuses alarms in GX and VRM.

Or did you have another solution?

Martin Jarvis ***@***.***> schrieb am Mo., 23. Sept. 2024,
10:58:

**Reply by freakent** [reply]
I'd have to go and read the dbus api documentation. How do you visualise the fuse status in VenusOS. Can you provide some screenshots?

**Comment by Parateam** [comment]
Plese read this post. This might be the solution. It is a battery device
with no volts and amps, but withe fuses.

Alarms will be shown in the right position.

https://community.victronenergy.com/t/lynx-distributor-protocol/3155/7

Martin Jarvis ***@***.***> schrieb am Mo., 23. Sept. 2024,
11:52:

**Reply by freakent** [reply]
I see. I think it would have made more sense if Victron had created a dedicated device for this, but that's not something we can change now so we'll have to e to work around it. 

I assume you have to add the device to dbus without adding all the regular battery parameters like voltage, current and SOC? 

I think the problem we would have currently is that I add all parameters for a device when it registers based on its device type. So I can't add lynx distributor parameters without adding voltage and SOC etc, otherwise someone else wouldn't able to battery devices in regular use cases. I have been thinking about splitting the services config file and allow multiple configs for each device type.

**Comment by Parateam** [comment]
Yes, i think voltage, current, soc could set to zero. The goal is to  check
Fuses and get an alarm.

Could you please add this? I dont know how much work this is for you.

Martin Jarvis ***@***.***> schrieb am Mo., 23. Sept. 2024,
13:04:
